### PR TITLE
test: add LSH clustering coverage for #975

### LINKED ISSUE #975: Tests: Add test coverage for mathematical clustering logic in `greedybear/cronjobs/commands/lsh.py`
I was looking at the test coverage recently and noticed that `greedybear/cronjobs/commands/lsh.py` currently has quite low coverage (around 21%). This file is really important since it contains the Locality-Sensitive Hashing (LSH) and Union-Find logic for sequence clustering. 

If you'd like to see exactly which lines need tests, you can run this command locally:
```bash
docker exec greedybear_uwsgi sh -c "coverage run --rcfile=.coveragerc manage.py test && coverage report -m greedybear/cronjobs/commands/lsh.py"
```
Getting this file closer to 100% test coverage would be a huge help to the project.

### PR DESCRIPTION
**Pull request by Demiserular**
# Description

This PR adds focused test coverage for the LSH clustering and Union-Find logic in greedybear/cronjobs/commands/lsh.py.

I added unit tests in tests/test_lsh.py to cover path compression, union behavior, MinHash signature generation, connected-component label compaction, empty input handling, clustering of similar sequences, and deterministic LSH query behavior with self-match filtering.

Targeted coverage for greedybear/cronjobs/commands/lsh.py is now 100%.

### Related issues

- Closes #975
- Reopens the same work after reassignment

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] Chore (refactoring, dependency updates, CI/CD changes, code cleanup, docs-only changes).

# Checklist

Please complete this checklist carefully. It helps guide your contribution and lets maintainers verify that all requirements are met.

### Formalities

- [ ] I have read and understood the rules about [how to Contribute](https://intelowlproject.github.io/docs/GreedyBear/Contribute/) to this project.
- [x] I chose an appropriate title for the pull request in the form: `<feature name>. Closes #999`
- [x] My branch is based on `develop`.
- [x] The pull request is for the branch `develop`.
- [x] I have reviewed and verified any LLM-generated code included in this PR.

### Docs and tests

- [ ] I documented my code changes with docstrings and/or comments.
- [x] I have checked if my changes affect user-facing behavior that is described in the [docs](https://intelowlproject.github.io/docs/GreedyBear/Introduction/). If so, I also created a pull request in the [docs repository](https://github.com/intelowlproject/docs).
- [ ] Linter (`Ruff`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowlproject.github.io/docs/GreedyBear/Contribute/#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [x] I have added tests for the feature/bug I solved.
- [x] All the tests gave 0 errors.

Validation notes:

- `python manage.py test` passed in Docker (`Ran 828 tests ... OK`).
- `ruff check .` currently reports existing migration-style issues outside this PR scope.

### GUI changes

Ignore this section if you did not make any changes to the GUI.

- [ ] I have provided a screenshot of the result in the PR.
- [ ] I have created new frontend tests for the new component or updated existing ones.

# Review process

- We encourage you to create a draft PR first, even when your changes are incomplete. This way you refine your code while we can track your progress and actively review and help.
- If you think your draft PR is ready to be reviewed by the maintainers, click the corresponding button. Your draft PR will become a real PR.
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem.
- Every time you make changes to the PR and you think the work is done, you should explicitly ask for a review. After receiving a "change request", address the feedback and click "request re-review" next to the reviewer's profile picture at the top right.

### CODE REVIEW

**Comment by Demiserular** [comment]
@regulartim  Addressed! Changed the base class to [django.test.TestCase](vscode-file://vscode-app/c:/Users/sc895/AppData/Local/Programs/Microsoft%20VS%20Code/cfbea10c5f/resources/app/out/vs/code/electron-browser/workbench/workbench.html) and added the missing edge cases for identical sequences, self unions, and single elements.

File: tests/test_lsh.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
+from unittest.mock import patch
+
+from django.test import TestCase
```
Comment (regulartim) [inline tests/test_lsh.py]: Please use SimpleTestCase or CustomTestCase like almost every other test does.

File: tests/test_lsh.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+        lsh = LSHConnectedComponents(threshold=0.7, num_perm=32)
+        sequences = [["a"], ["b"], ["c"]]
+
+        with (
+            patch.object(lsh, "_get_min_hashes", return_value=["mh0", "mh1", "mh2"]),
+            patch("greedybear.cronjobs.commands.lsh.MinHashLSH", FakeLSH),
+        ):
+            labels = lsh.get_components(sequences)
+
```
Comment (regulartim) [inline tests/test_lsh.py]: This test mocks both _get_min_hashes and MinHashLSH, essentially re-implementing the get_components algorithm. So what is actually tested here? test_get_components_groups_similar_sequences already covers the tested wiring more meaningfully, right?

**Comment by Demiserular** [comment]
> Hey @Demiserular ! Looks good all in all, but I wonder why you are using TestCase here...

TestCase is the standard default class for writing tests in Django. that's why

File: tests/test_lsh.py (hunk 2 above)
Comment (Demiserular) [reply tests/test_lsh.py]: "My original intent with the mock was to write a strict, isolated unit test to verify just the internal loop and [UnionFind](vscode-file://vscode-app/c:/Users/sc895/AppData/Local/Programs/Microsoft%20VS%20Code/cfbea10c5f/resources/app/out/vs/code/electron-browser/workbench/workbench.html) logic, separated entirely from the actual datasketch math computations.

However, after looking at it again, I completely agree with you. Testing the wiring in complete isolation here doesn't add much value, and [test_get_components_groups_similar_sequences](vscode-file://vscode-app/c:/Users/sc895/AppData/Local/Programs/Microsoft%20VS%20Code/cfbea10c5f/resources/app/out/vs/code/electron-browser/workbench/workbench.html) indeed covers the actual end-to-end flow much more meaningfully.

I've gone ahead and removed the redundant mock test to keep things clean. (Also swapped TestCase to [SimpleTestCase](vscode-file://vscode-app/c:/Users/sc895/AppData/Local/Programs/Microsoft%20VS%20Code/cfbea10c5f/resources/app/out/vs/code/electron-browser/workbench/workbench.html)). Updates are pushed!
